### DISCUSSION TITLE
blog/2020/12/03/ipw-tscs-msm/index

### DISCUSSION THREAD

**Discussion by giscus** [andrewheiss/ath-quarto#66] [Blog comments]
# blog/2020/12/03/ipw-tscs-msm/index

Use R to close backdoor confounding in panel data with marginal structural models and inverse probability weights for both binary and continuous treatments

https://www.andrewheiss.com/blog/2020/12/03/ipw-tscs-msm/index.html

**Comment by ghost** [comment]
Thank you for this post! I'm an epidemiology student messing around with panel data, so seeing that the models I'm familiar with are (or can be) used in these contexts is nice to see. I'm a little overwhelmed by how many models and estimators there seem to be in the econometrics literature and the only thing I think I'm sure about that TWFE is usually a bad idea!

I know this post is a little older now, but I think your issue had to do with the data you simulated. I tried replacing the policy-variable (your binary treatment) with a random continuous variable before simulating outcomes (so that policy becomes a continuous treatment). I then used the MSM-code you provided for a continuous treatment, changing the models to fit your data-generating process for policy, and what I ended up with was an estimated effect of ~7 (7.09, to be exact). That seems correct, right? Good news for me, because I was looking for an example of MSMs being used for panel data and this post is exactly that. :)
